src/preload/client/journals.ts

```diff
@@ -1,83 +1,125 @@
 
 export type IJournalsClient = JournalsClient;
 
 export class JournalsClient {
-  constructor(private db: Database) {}
+  constructor(
+    private db: Database,
+    private files: IFilesClient,
+    private preferences: IPreferencesClient,
+  ) {}
 
   list = async (): Promise<JournalResponse[]> => {
-    return this.db.prepare("select * from journals order by name").all();
+    const journals = this.db
+      .prepare("select * from journals order by name")
+      .all();
+
+    const archived = await this.preferences.get("ARCHIVED_JOURNALS");
+    journals.forEach((j) => {
+      j.archived = archived[j.name];
+    });
+
+    return journals;
   };
 
-  create = (journal: { name: string }): Promise<JournalResponse> => {
+  create = async (journal: { name: string }): Promise<JournalResponse> => {
     const name = validateJournalName(journal.name);
-    const id = uuidv7();
+    await this.files.createFolder(name);
+    return this.index(name);
+  };
+
+  index = async (journalName: string): Promise<JournalResponse> => {
+    // add to archived if not already present
+    const existing = await this.preferences.get(
+      `ARCHIVED_JOURNALS.${journalName}`,
+    );
+    if (existing == null) {
+      await this.preferences.set(`ARCHIVED_JOURNALS.${journalName}`, false);
+    }
 
     this.db
       .prepare(
-        "insert into journals (id, name, createdAt, updatedAt) values (:id, :name, :createdAt, :updatedAt)",
+        "insert into journals (name, createdAt, updatedAt) values (:name, :createdAt, :updatedAt)",
       )
       .run({
-        id,
-        name,
+        name: journalName,
         createdAt: new Date().toISOString(),
         updatedAt: new Date().toISOString(),
       });
 
-    return this.db.prepare("select * from journals where id = :id").get({ id });
+    return this.db
+      .prepare("select * from journals where name = :name")
+      .get({ name: journalName });
   };
 
-  update = (journal: {
-    id: string;
-    name: string;
-  }): Promise<JournalResponse> => {
-    const name = validateJournalName(journal.name);
+  rename = async (
+    // note: do not pass the full JournalResponse mobx object here; restructure
+    // it to just the name and archived fields
+    journal: { name: string; archived: boolean },
+    newName: string,
+  ): Promise<JournalResponse> => {
+    newName = validateJournalName(newName);
+    await this.files.renameFolder(journal.name, newName);
 
     this.db
       .prepare(
-        "update journals set name = :name, updatedAt = :updatedAt where id = :id",
+        "update journals set name = :newName, updatedAt = :updatedAt where name = :name",
       )
       .run({
-        name,
-        id: journal.id,
+        name: journal.name,
+        newName,
         updatedAt: new Date().toISOString(),
       });
 
+    // todo: dumb; revert to having documents reference id instead of name
+    this.db
+      .prepare("update documents set journal = :newName where journal = :name")
```

Comment:
Example of where changing the reference to be by name results in extra work; obviously dumb design here. But works for now....